validator/lib/validateTileset.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var Promise = require('bluebird');
+
+module.exports = validateTileset;
+
+/**
+ * Walks down the tree represented by the JSON object and checks if it is a valid tileset.
+ *
+ * @param {Object} tileset The JSON object representing the tileset.
+ * @return {Promise} A promise that resolves with two parameters - (1) a boolean for whether the tileset is valid
+ *                                                                 (2) the error message if the tileset is not valid.
+ *
+ */
+function validateTileset(tileset) {
+    return new Promise(function(resolve) {
+        validateNode(tileset.root, tileset.geometricError, resolve);
+    });
+}
+
+function validateNode(root, parentGeometricError, resolve) {
+    var stack = [];
+    stack.push({'node': root, 'parentError': parentGeometricError});
+
```

Comment:
Instead of adding the parent error, add the parent. This will be useful for validation besides geometric error.

Remove the quotes and format to multiple lines instead:

```javascript
stack.push({
    tile : root,
    parent : undefined
});
```